core/src/main/java/org/mule/processor/chain/NestedProcessorChain.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.processor.chain;
+
+import org.mule.DefaultMuleEvent;
+import org.mule.DefaultMuleMessage;
+import org.mule.api.MuleContext;
+import org.mule.api.MuleEvent;
+import org.mule.api.MuleMessage;
+import org.mule.api.NestedProcessor;
+import org.mule.api.context.MuleContextAware;
+import org.mule.api.processor.MessageProcessor;
+
+import java.util.Map;
+
+/**
+ * An implementation of {@link NestedProcessor} that
+ * wraps a {@link MessageProcessor} and allows to
+ * execute it
+ *
+ * @since 3.7.0
+ */
+public class NestedProcessorChain implements NestedProcessor, MuleContextAware
```

Comment:
This class actually belongs to devkit-support.. Devkit guys wrote it... I just moved it here (and deprecated the devkit one) to be able to use it without depending on devkit-support.

As we move to a NB environment, I plan to stop using this class and send it back home.